src/conversation/mod.rs

```diff
@@ -245,3 +252,29 @@ impl<D: RepositoryEmbeddingsDB, M: EmbeddingsModel> Conversation<D, M> {
         }
     }
 }
+
+async fn sanitize_query(query: &str) -> Result<String> {
+    let message = ChatCompletionMessage {
+        name: None,
+        function_call: None,
+        role: MessageRole::user,
+        content: sanitize_query_prompt(query),
+    };
+    let client = Client::new(env::var("OPENAI_API_KEY")?);
+    let request = generate_completion_request(vec![message], "none");
+    let response = client.chat_completion(request).await?;
```

Comment:
I think this is a clever solution: use the OpenAI api's to sanitize the query. But it may be rather expensive (given this essentially doubles all queries to the OpenAI apis).

A far off future optimization may be to have some "on server" process that sanitizes queries from clients.

Reply:
Right. This adds an additional API call to for every query. Your idea to have our own process can be worked out. I believe any small size llm can perform this sanitization.